File: dotnetcore/core/Platform/Data/History.cs
namespace Platform.Data
{
    using System.Collections.Generic;

    public class History : BaseEntity
    {
        //Id为[ClassName]_[Id]

        public List<HistoryInfo> HistoryInfos;

        public History(string id, List<HistoryInfo> historyInfos)
        {
            this.Id = id;
            this.HistoryInfos = historyInfos;
        }
    }
}

File: java/core/platform/src/main/java/com/wiiee/core/platform/history/History.java
package com.wiiee.core.platform.history;

import com.wiiee.core.platform.data.BaseData;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * Created by wiiee on 9/1/2017.
 */
public class History extends BaseData<String> {
    private List<HistoryInfo> historyInfos;

    public History() {

    }

    public History(String id) {
        super(id);
        this.historyInfos = new ArrayList<>();
    }

    public History(String id, HistoryInfo info) {
        super(id);
        this.historyInfos = Arrays.asList(info);
    }

    public List<HistoryInfo> getHistoryInfos() {
        return historyInfos;
    }

    public void setHistoryInfos(List<HistoryInfo> historyInfos) {
        this.historyInfos = historyInfos;
    }

    public void addHistoryInfo(HistoryInfo info) {
        historyInfos.add(info);
    }
}
